telemetry/prometheus.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package telemetry
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const namespace = "thor_telemetry"
+
+// InitializePrometheusTelemetry creates a new instance of the Prometheus service and
+// sets the implementation as the default telemetry services
+func InitializePrometheusTelemetry() {
+	// don't allow for reset
+	if _, ok := telemetry.(*prometheusTelemetry); !ok {
+		telemetry = newPrometheusTelemetry()
+	}
+}
+
+type prometheusTelemetry struct {
+	counters      sync.Map
+	counterVecs   sync.Map
+	histograms    sync.Map
+	histogramVecs sync.Map
+	gaugeVecs     sync.Map
+	gauges        sync.Map
+}
+
+func newPrometheusTelemetry() Telemetry {
+	return &prometheusTelemetry{
+		counters:      sync.Map{},
```

Comment:
Look like summary is missing, do you think it will be used in the near future?

Reply:
Summaries are somewhat expensive. I was considering that we would add them when the need for them arises ?

Reply:
Sure